blacklistfilter/blacklist_downloader/bl_downloader.py.in

```diff
@@ -131,11 +132,26 @@ class Blacklist:
             if req.status_code == 200:
                 data = req.content
 
-                # If CSV, cut the desired column
-                with suppress(AttributeError):
-                    data = self.cut_csv(data)
+                # JSON
+                try:
+                    data = json.loads(data)  # if not JSON, throws exception
+
+                    data_list = []
+
+                    for i in data:
+                        ip_val = i.get('address')
```

Comment:
Hard-coded key name? That way, it only supports JSON from Smashed, which is not very good design.

The key to look for should be a part of configuration. Whether to parse it as JSON or CSV should be there as well.

Reply:
Config file has been expanded to include these attributes.